content/install_maintain/installation/installing-external-db-and-queue-cluster.md

```diff
@@ -0,0 +1,215 @@
+---
+layout: bt_wiki
+title: Installing and Configuring External DB And Rabbitmq Within Distributed Cluster
+description: Install and Configure External DB And Rabbitmq within Distributed Cluster . 
+category: Installation
+draft: false
+weight: 5
+---
+
+
+# Installing and Configuring External DB And Rabbitmq Within Distributed Cluster
+
+When installing the Cloudify cluster, the user can use externally PostgreSQL database and RabbitMQ.
+This page is a guide for installing such services.
+
+### PostgreSQL Database Cluster
+
+The PostgreSQL database high-availability cluster is comprised of 3 nodes (Cloudify best-practice) or more.
+
+**Note** Make sure the following ports are open for each node:
+
+ Port      | Description
+-----------|------------
+ tcp/2379  | etcd port.
+ tcp/2380  | etcd port.
+ tcp/5432  | PostgreSQL connection port.
+ tcp/8008  | Patroni control port.
+
+
+#### Externally Hosted PostgreSQL Database Installation
+ - Make sure the PostgreSQL instance is publicly available and reachable from the local Cloudify Management service cluster nodes.
+ - Retrieve the PostgreSQL instance CA certificate and save it locally for future use in the Cloudify Management service cluster nodes configuration.
+ - Keep your PostgreSQL database username and password for the later configuration of the Cloudify Management service cluster nodes. 
+
+##### Azure DBaaS for PostgreSQL
+
+Cloudify supports [Microsoft's Azure Database for PostgreSQL](https://docs.microsoft.com/en-us/azure/postgresql/) as an external database option replacing Cloudify's PostgreSQL deployment.  
+
+Azure Database for PostgreSQL is a fully managed database-as-a-service offering that can handle mission-critical workloads with predictable performance, security, high availability, and dynamic scalability. It is available in two deployment options, as a single server and as a Hyperscale (Citus) cluster (preview).  
+
+###### Setting up Azure database for PostgreSQL as the Cloudify database  
+The DBaaS of Azure supports a clustered instance and a single instance available for resizing on demand.  
+As opposed to other DBaaS vendors, Azure doesn't give access to the `postgres` user with SuperUser privileges, so while working with Azure DBaaS is fully supported, the configuration is a bit different than regular Postgres installations.  
+
+Using Azure DBaaS (either the single instance or the clustered instance), requires specific setup changes to the Cloudify manager configuration.    
+Azure connection string for the users must be in the form of `<username>@<dbhostname>`, so for a DB user named `cloudify` and a db hostname named `azurepg`, the user that needs to be configured should be: `cloudify@azurepg`.  
```

Comment:
does it mean that the user need to be created when provisioning DB in Azure?

Reply:
The master user is created by azure with any name the user wants(its an input in the creation path). 
later cloudify creates another user for connection this db